ui/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -823,6 +829,22 @@ export default class ConfirmTransactionBase extends Component {
       txData.contractMethodName = name;
     }
 
+    if (dappProposedTokenAmount) {
+      txData.dappProposedTokenAmount = dappProposedTokenAmount;
+      txData.originalApprovalAmount = dappProposedTokenAmount;
+    }
+
+    if (customTokenAmount) {
+      txData.customTokenAmount = customTokenAmount;
+      txData.finalApprovalAmount = customTokenAmount;
+    } else {
+      txData.finalApprovalAmount = dappProposedTokenAmount;
+    }
+
+    if (currentTokenBalance) {
+      txData.currentTokenBalance = currentTokenBalance;
+    }
+
```

Comment:
```txData.dappProposedTokenAmount = dappProposedTokenAmount;``` this line seems unnecessary. .. dappProposedTokenAmount is coming *from* txData we don't need to add it back. I get the need to add final/original here but any of the values we are just getting from txData *do not need to be added back to txData because they were retrieved from their originally* unless i'm mistaken.

Reply:
@brad-decker I understand what are you saying, but I think the name of this props (`dappProposedTokenAmount`, `customTokenAmount` and  `currentTokenBalance`) is what confusing you here, because I used the same names to create properties in `txData`. 

In `confirm-approve.js` I passed the values for this props in `ConfirmTransactionBase` container like this:

> customTokenAmount={String(customPermissionAmount)}
    dappProposedTokenAmount={tokenAmount}
    currentTokenBalance={tokenBalance}

So like @danjm said in https://github.com/MetaMask/metamask-extension/issues/15152#issuecomment-1177605611:

> `customTxParamsData` is passed from `ConfirmApprove` to `confirm-transaction-base.container`, we somehow also need to passed the original amount (in this case `dappProposedTokenAmount`) and chosen amount (in this case `customTokenAmount`) to the container and then the component so that it can be passed as part of `txMeta` in `handleSubmit` in `ConfirmTransactionBase` component.

Also we should also put `currentTokenBalance` on the `txData`, which should be the balance of the token that is being send (in this case `currentTokenBalance`), and should match the balance that is rendered in the edit permission modal on the approve screen, this can then be used in the controller to calculate `transaction_approval_amount_vs_balance_ratio`.

For example, if we don't do `txData.dappProposedTokenAmount = dappProposedTokenAmount;` in `handeSubmit` in `ConfirmTransactionBase` component, in transaction controller `dappProposedTokenAmount` property from `txMeta` will be `undefined`. The same applies to `customTokenAmount` and `currentTokenBalance`.

So I think the name of this props is confusing, I can rename it, let's say like this:

> customAmount={String(customPermissionAmount)}
    originalAmount={tokenAmount}
    tokenBalance={tokenBalance}

and then it will be `txData.dappProposedTokenAmount = originalAmount;`, `txData.customTokenAmount = customAmount;` and `txData.currentTokenBalance = tokenBalance;`

If this is okay to you or if you have some other questions, please comment and we will discuss about that.

Reply:
Again I am super confused. txData is the state of the transaction in the background. 

```js
  const {
    txParams = {},
    id: transactionId,
    type,
    customTokenAmount,
    currentTokenBalance,
    dappProposedTokenAmount,
  } = txData;
```
these lines imply that customTokenAmount, currentTokenBalance and dappProposedTokenAmount *already exist on the txData object* -- if they are here, then they do not need to be added to it again. Furthermore you're passing these as props to ConfirmTransactionBase **AND** specifying them in the container to get them from the txData object in state..so where in the confirmation-transaction-base.component.js file are the props coming from? From the confirm-approve file or the .container file? 

I pulled this locally and removed all of the changes to the .container file because these values are being passed from the confirm-approve component, not from state. It worked without the changes to the .container file. Could you experiment and see if you have the same results?

Reply:
@brad-decker Props coming from the `confirm-approve` file. These values are being passed from the `confirm-approve` component and I tried the same as you, it worked without the changes to the .container file. So that changes is .container file I will remove, but what about the changes in `confirmation-transaction-base.component.js`? Is it okay to stay like it is now? What are you suggesting?
Sorry for misunderstanding.

Reply:
We can keep the changes in the component file. The reason I was confused is because we were destructuring out of a state key in the container file.  This misdirection implied the existence of the values on the txData *already* so thats why I was confused.

Reply:
@brad-decker Yeah, sorry for that. I will make changes now. Thank you for explanation, I understand you in first, but it seems I didn't explain it well enough for you to understand. Sorry for that again.

Reply:
No harm done @VSaric! Thank you 🙏🏻

Reply:
@brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/15289/commits/546ec4254f645b25569630603b05fe1cb015b9a1).